lib/chef/resource/windows_security_policy.rb

```diff
@@ -83,6 +84,59 @@ class WindowsSecurityPolicy < Chef::Resource
       description: "Policy value to be set for policy name."
 
       load_current_value do |desired|
+        current_state = load_security_options
+
+        if desired.secoption == "ResetLockoutCount"
+          if desired.secvalue.to_i > 30
+            raise Chef::Exceptions::ValidationFailed, "The \"ResetLockoutCount\" value cannot be greater than 30 minutes"
+          end
+        end
+        if desired.secoption == "ResetLockoutCount" || desired.secoption == "LockoutDuration"
+          if current_state["LockoutBadCount"] == "0"
+            raise Chef::Exceptions::ValidationFailed, "#{desired.secoption} cannot be set unless the \"LockoutBadCount\" security policy has been set to a non-zero value"
+          else
+            secvalue current_state[desired.secoption.to_s]
+          end
+        else
+          secvalue current_state[desired.secoption.to_s]
+        end
+      end
+
+      action :set do
+        converge_if_changed :secvalue do
+          security_option = new_resource.secoption
+          security_value = new_resource.secvalue
+
+          file = Tempfile.new(["#{security_option}", ".inf"])
+          case security_option
+          when "LockoutBadCount"
+            cmd = "net accounts /LockoutThreshold:#{security_value}"
+          when "ResetLockoutCount"
+            cmd = "net accounts /LockoutWindow:#{security_value}"
+          when "LockoutDuration"
+            cmd = "net accounts /LockoutDuration:#{security_value}"
+          when "NewAdministratorName", "NewGuestName"
+            policy_line = "#{security_option} = \"#{security_value}\""
+            file.write("[Unicode]\r\nUnicode=yes\r\n[System Access]\r\n#{policy_line}\r\n[Version]\r\nsignature=\"$CHICAGO$\"\r\nRevision=1\r\n")
+            file.close
+            file_path = file.path.gsub("/", '\\')
+            cmd = "C:\\Windows\\System32\\secedit /configure /db C:\\windows\\security\\new.sdb /cfg #{file_path} /areas SECURITYPOLICY"
+          else
+            policy_line = "#{security_option} = #{security_value}"
+            file.write("[Unicode]\r\nUnicode=yes\r\n[System Access]\r\n#{policy_line}\r\n[Version]\r\nsignature=\"$CHICAGO$\"\r\nRevision=1\r\n")
+            file.close
+            file_path = file.path.gsub("/", '\\')
+            cmd = "C:\\Windows\\System32\\secedit /configure /db C:\\windows\\security\\new.sdb /cfg #{file_path} /areas SECURITYPOLICY"
+          end
+          powershell_out!(cmd)
+          file.unlink
```

Comment:
```suggestion
          Tempfile.new(["#{security_option}", ".inf"]) do |file|
            case security_option
            when "LockoutBadCount"
              cmd = "net accounts /LockoutThreshold:#{security_value}"
            when "ResetLockoutCount"
              cmd = "net accounts /LockoutWindow:#{security_value}"
            when "LockoutDuration"
              cmd = "net accounts /LockoutDuration:#{security_value}"
            when "NewAdministratorName", "NewGuestName"
              policy_line = "#{security_option} = \"#{security_value}\""
              file.write("[Unicode]\r\nUnicode=yes\r\n[System Access]\r\n#{policy_line}\r\n[Version]\r\nsignature=\"$CHICAGO$\"\r\nRevision=1\r\n")
              file_path = file.path.gsub("/", '\\')
              cmd = "C:\\Windows\\System32\\secedit /configure /db C:\\windows\\security\\new.sdb /cfg #{file_path} /areas SECURITYPOLICY"
            else
              policy_line = "#{security_option} = #{security_value}"
              file.write("[Unicode]\r\nUnicode=yes\r\n[System Access]\r\n#{policy_line}\r\n[Version]\r\nsignature=\"$CHICAGO$\"\r\nRevision=1\r\n")
              file_path = file.path.gsub("/", '\\')
              cmd = "C:\\Windows\\System32\\secedit /configure /db C:\\windows\\security\\new.sdb /cfg #{file_path} /areas SECURITYPOLICY"
            end
            powershell_out!(cmd)
          end
```

Reply:
so, I tested this, and `powershell_out` and `powershell_exec` both stopped applying the changes in the cmd.  Not sure why that would be, if there's a bug in putting shellouts in a wrapper like that or something, but it definitely stopped actually running the shellouts on the system.

Reply:
Huh. It could be a buffering issue or something like that, but I understand if you'd rather just move on. 👍

Reply:
If you're curious, I just ran into this myself. `Tempfile.new { }` will not call the block, `Tempfile.create { }` will. 🤦